ports/nrf/modules/machine/pin.c

```diff
@@ -215,8 +215,22 @@ const pin_obj_t *pin_find(mp_obj_t user_obj) {
 STATIC void pin_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t kind) {
     pin_obj_t *self = self_in;
 
-    mp_printf(print, "Pin(%d, mode=%s)", self->pin,
-            (nrf_gpio_pin_dir_get(self->pin) == NRF_GPIO_PIN_DIR_OUTPUT) ? "OUT" : "IN");
+    char *pull = "PULL_DISABLED";
+    switch (nrf_gpio_pin_pull_get(self->pin)) {
+        case NRF_GPIO_PIN_PULLUP:
+            pull = "PULL_UP";
+            break;
+        case NRF_GPIO_PIN_PULLDOWN:
+            pull = "PULL_DOWN";
+            break;
+        case NRF_GPIO_PIN_NOPULL:
```

Comment:
I'm kind of missing a default statement in this switch, could this be the one? Instead of testing against NOPULL and break anyway?

Reply:
Hmmm maybe that's a little bit neater. I've updated the PR.
(I only added the third `case` because the compiler was complaining that I didn't check all the possible values).